docs/submitting_data.rst

```diff
@@ -1,9 +1,122 @@
 Submitting Data
 ===============
 
-Using the Python Client
+To submit your test data to Treeherder, you have two options:
+
+1. :ref:`submitting-using-pulse`
+
+    This is the new process Task Cluster is using to submit data to Treeherder.
+    There is a `YML Schema`_ to validate your payload against to ensure it will
+    be accepted.  In this case, you create your own `Pulse`_ exchange and publish
+    to it.  To get Treeherder to receive your data, you would create a bug to
+    have your Exchange added to Treeherder's config.  All Treeherder instances
+    can subscribe to get your data, as can local dev instances for testing.
+
+    While it is beyond the scope of this document to explain how `Pulse`_ and
+    RabbitMQ work, we encourage you to read more about this technology on
+    its Wiki page.
+
+2. :ref:`submitting-using-python-client`
+
+    This is historically how projects and users have submitted data to Treeherder.
+    This requires getting Hawk credentials approved by a Treeherder Admin.
+    There is a client library to help make this easier.  However, there is no
+    schema to validate the payload against.  But using the client to build your
+    payload will help you get it in the accepted form.  Your data only goes to
+    the host you send it to.  Dev instances can not subscribe to this data.
+
+
+.. _submitting-using-pulse:
+
+Submitting using Pulse
+======================
+
+To submit via a Pulse exchange, these are the steps you will need to follow:
+
+1. Format your data
+-------------------
+
+The new job format in `YML Schema`_ is significantly different from the one
+used in our API job submission.  It has virtually the same data, just in
+(what we hope) is a better form.  You are responsible for validating your data
+prior to publishing it onto your Exchange, or Treeherder may reject it.
```

Comment:
I would not bother contrasting the pulse data with the API data here. Also I would say `Pulse Job Schema` as opposed to `YML Schema`.

Maybe:

```
You should format your job data according to the `Pulse Job Schema`_, which describes the various properties of a job: whether it passed or failed, job group/type symbol, description, log information, etc. You are responsible for validating your data prior to publishing it onto your exchange, or Treeherder may reject it.
```